src/meta/src/hummock/manager/compaction/compaction_group_schedule.rs

```diff
@@ -559,18 +559,13 @@ impl HummockManager {
             .into_iter()
             .for_each(|task_assignment| {
                 if let Some(task) = task_assignment.compact_task.as_ref() {
-                    let input_sst_ids: HashSet<u64> = task
-                        .input_ssts
-                        .iter()
-                        .flat_map(|level| level.table_infos.iter().map(|sst| sst.sst_id))
-                        .collect();
-                    let input_level_ids: Vec<u32> = task
-                        .input_ssts
-                        .iter()
-                        .map(|level| level.level_idx)
-                        .collect();
-                    let need_cancel = !levels.check_sst_ids_exist(&input_level_ids, input_sst_ids);
-                    if need_cancel {
+                    let is_compaction_group_version_id_match = levels.compaction_group_version_id
+                        == task_assignment
```

Comment:
We can encapsulate a method belonging to compact-task for expire judgment

```rust
impl CompactTask {
  pub fn is_expired(&self, version_id) {
    self.version_id >= version_id
  }
}
```

Strictly speaking, there's no need for an absolute match.